CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 3.23)
+cmake_minimum_required(VERSION 3.30)
```

Comment:
Is there a feature you used that required this version upgrade? I'd like it if we could keep this version as low as possible. 

From my own selfish (ROS2) perspective, 3.23 already breaks the build for ubuntu 22.04 (which has CMake v3.22) , and anything higher than 3.28 will break it for 24.04.    One can always go and install more up-to-date CMake versions using kitware's PPA, but it's just another barrier to entry I'd like to avoid, if possible.